### DISCUSSION TITLE
Why is the forward pass compute for transpose a no-op?

### DISCUSSION THREAD

**Discussion by wilderfield** [ggml-org/llama.cpp#6221] [Q&A] [answered]
https://github.com/ggerganov/llama.cpp/blob/fa046eafbc70bf97dcf39843af0323f19a8c9ac3/ggml.c#L11646-L11654

**Comment by ggerganov** [comment, marked as answer]
We just swap the strides `nb[0]` and `nb[1]` without moving the actual data in memory
